src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```diff
@@ -1014,16 +1015,26 @@ public int getMaxColumnsInTable() throws SQLServerException {
     @Override
     public int getMaxConnections() throws SQLException, SQLTimeoutException {
         checkClosed();
+        SQLServerResultSet rs = null;
         try {
-            String s = "sp_configure 'user connections'";
-            SQLServerResultSet rs = getResultSetFromInternalQueries(null, s);
-            if (!rs.next())
+            rs = getResultSetFromInternalQueries(null,
```

Comment:
`rs` is never closed.

Reply:
I noticed `getUserName()` leaks a `ResultSet` too. Would you also double check this class for other leaks?

Reply:
`getUserName()` is fine, its in a try-with-resources block. I fixed this one, good catch!